client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -289,6 +302,26 @@ public StreamInfo getStreamInfo(String scopeName, String streamName) {
                                 });
     }
 
+    @Override
+    public <T> CompletableFuture<T> fetchEvent(EventPointer pointer, Serializer<T> serializer) {
+        Preconditions.checkNotNull(pointer);
+        Preconditions.checkNotNull(serializer);
+        CompletableFuture<T> completableFuture = CompletableFuture.supplyAsync(() -> {
+            @Cleanup
+            EventSegmentReader inputStream = inputStreamFactory.createEventReaderForSegment(pointer.asImpl().getSegment(), pointer.asImpl().getEventLength());
+            inputStream.setOffset(pointer.asImpl().getEventStartOffset());
+            try {
+                ByteBuffer buffer = inputStream.read();
+                return  serializer.deserialize(buffer);
+            } catch (EndOfSegmentException e) {
+                throw Exceptions.sneakyThrow(new NoSuchEventException(e.getMessage()));
+            } catch (NoSuchSegmentException | SegmentTruncatedException e) {
+                throw Exceptions.sneakyThrow(new NoSuchEventException("Event no longer exists."));
```

Comment:
This is not consistent with what the documentation says. Either change the documentation or the implementation.

Reply:
yeah changed The documentation.